### ISSUE TITLE
Seems there is a NaN losses in the baseline

### ISSUE THREAD

**Issue by LLMforScience** [ElliottYan/LUFFY#24]
I tried your script `train_rl_sft_loss.sh `and set the weight of pg_loss to zero, which means I'm only training with SFT loss.

However, the SFT loss doesn't decrease resulting NaN finally, and the performance is very poor.

Could you please help me understand why this is happening? Thanks!

<img width="1800" height="643" alt="Image" src="https://github.com/user-attachments/assets/f01487d1-6aae-487a-9311-96db87a55809" />

**Comment by ElliottYan** [comment]
Hi, I haven't tried this approach yet. Can you provide your script so that I can reproduce your issue?

**Comment by yafuly** [comment]
For SFT, we suggest using the SFT training script (based on Open-RLHF) directly to reproduce our results.

**Comment by yafuly** [comment]
"I also directly tried you script train_rl_sft_loss.sh without any modification. The losses (Both PG and SFT) are NaN."

Could you provide more details to reproduce this error? e.g., datasets, base LLMs, etc.

**Comment by yuanyige** [comment]
<img width="1022" height="397" alt="Image" src="https://github.com/user-attachments/assets/f9a80fa9-3903-48df-8e18-6f45d67d02a1" />

**Comment by ElliottYan** [comment]
It looks like a problem caused by the re-batching strategy of verl. For ppo update, all the generated data would be shuffled,  batched, and updated.
It is possible that in a batch, there is no off-policy data, causing NaN when computing the average. 
https://github.com/ElliottYan/LUFFY/blob/main/luffy/verl/verl/mix_src/mix_core_alg.py#L9

This would lead to SFT loss to be nan, but it would **not affect the actual gradient computation**, only the visualization. 

<img width="792" height="356" alt="Image" src="https://github.com/user-attachments/assets/355db1d3-0475-472a-8b5c-b18db82e73c0" />

**Comment by ElliottYan** [comment]
Also, poor performance of multi-task sft loss at the early stage of RL training is expected. If you run more steps, the performance would increase. 

> I tried your script `train_rl_sft_loss.sh `and set the weight of pg_loss to zero, which means I'm only training with SFT loss.
> 
> However, the SFT loss doesn't decrease resulting NaN finally, and the performance is very poor.
> 
> Could you please help me understand why this is happening? Thanks!
> 
> <img alt="Image" width="1800" height="643" src="https://private-user-images.githubusercontent.com/83741666/465848192-f01487d1-6aae-487a-9311-96db87a55809.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2NDgxNDQsIm5iZiI6MTc1MjY0Nzg0NCwicGF0aCI6Ii84Mzc0MTY2Ni80NjU4NDgxOTItZjAxNDg3ZDEtNmFhZS00ODdhLTkzMTEtOTZkYjg3YTU1ODA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE2VDA2MzcyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0NjliMWMyYTkwN2RhODMxNjUzMmQ0NjIxZGM1MDk0ZGY2N2U4ZmZjNjEzMjRjOGQ5MWFkMjZmMWVkMmRlZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8jaMJcVWM3-Bsw0jZ13EO5nZ9TpLl2S0L8DjqwoA_rg">
